sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -432,8 +452,17 @@ export class BatchingReceiverLite {
 
       try {
         const data: ServiceBusMessageImpl = this._createServiceBusMessage(context);
-        if (brokeredMessages.length < args.maxMessageCount) {
-          brokeredMessages.push(data);
+        brokeredMessages.push(data);
+
+        // NOTE: we used to actually "lose" any extra messages. At this point I've fixed the areas that were causing us to receive
+        // extra messages but if this bug arises in some other way it's better to return the message than it would be to let it be
+        // silently dropped on the floor.
+        if (brokeredMessages.length > args.maxMessageCount) {
+          logger.error(
```

Comment:
Should this be logger.warning? I think `error` is meant to be reserved for process ending events.

Reply:
You're 100% correct.

Reply:
Fixed